### PR TITLE
add max fs watch limit

### PR DESCRIPTION
**Pull request by sandikodev**
for several error's kinda like 
```
Error: ENOSPC: System limit for number of file watchers reached, watch '/__w/try-deploy-gayatri/try-deploy-gayatri/node_modules/@babel/plugin-transform-react-jsx-self/node_modules/@babel/core/lib/transformation'
    at FSWatcher.<computed> (internal/fs/watchers.js:218:26)
    at Object.watch (fs.js:1525:34)
    at NodeWatcher.watchdir (/__w/try-deploy-gayatri/try-deploy-gayatri/node_modules/sane/src/node_watcher.js:159:22)
    at Walker.<anonymous> (/__w/try-deploy-gayatri/try-deploy-gayatri/node_modules/sane/src/common.js:109:31)
    at Walker.emit (events.js:315:20)
    at /__w/try-deploy-gayatri/try-deploy-gayatri/node_modules/walker/lib/walker.js:69:16
    at FSReqCallback.oncomplete (fs.js:171:23) {
  errno: -28,
  syscall: 'watch',
  code: 'ENOSPC',

```

just puting them
```
  RUN echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
  RUN cat /proc/sys/fs/inotify/max_user_watches
```
on `Dockerfile`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -64,6 +64,9 @@ RUN curl -sS -L https://github.com/facebook/buck/releases/download/v${BUCK_VERSI
 
 # Full reference at https://dl.google.com/android/repository/repository2-1.xml
 # download and unpack android
+RUN echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
```
Comment (gengjiawen) [inline Dockerfile:67]: Can you join those two lines in one ?

Comment (gengjiawen) [reply Dockerfile:67]: Also, the command looks not working

```
sysctl: setting key "fs.inotify.max_user_watches": Read-only file system
The command '/bin/sh -c echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p' returned a non-zero code: 255
```

**Comment by gengjiawen** [comment]
Maybe just use `RUN echo 'fs.inotify.max_user_watches=524288' >> /etc/sysctl.conf`

https://github.com/laradock/laradock/blob/5b321d6fd71646d5aeecea2087bc420d50559d1a/ide-theia/Dockerfile#L7

File: Dockerfile (hunk 1 above)
Comment (meramsey) [reply Dockerfile]: that only works on the host OS fought that same battle before. 
Whatever is running docker engine should have that done.

Temporary:
```bash
sudo sysctl fs.file-max=2097152
sudo sysctl fs.inotify.max_user_watches=524288
sudo sysctl -p
```

Permanent
```bash
echo 'fs.file-max=2097152' | sudo tee -a /etc/sysctl.conf
echo 'fs.inotify.max_user_watches=524288' | sudo tee -a /etc/sysctl.conf
sudo sysctl -p
```

Try the above on where docker is installed and rerun should then work.

**Comment by cortinico** [comment]
Can we close this? I believe I haven't seen such errors in a long time

**Comment by cortinico** [comment]
Hey, I'm closing this PR as it appears to be stale.
Feel free to re-open it or create a new PR if the problem persists.

**Pull request closed without merging**